### PR TITLE
ant: fixing data races

### LINKED ISSUE #6186: Data races in `ant` module
### Describe the bug

When running `5_1_grt` stage for Ibex, ThreadSanitizer detects a few data races, usually in `ant::AntennaChecker`.

### Expected Behavior

Successful run without data races

### Environment

```shell
[WARNING] Your current OpenROAD version is outdated.
It is recommened to pull the latest changes.
If problem persists, file a github issue with the re-producible test case.
kernel: Linux 6.11.5-arch1-1
os: Arch Linux
cmake version 3.30.5
-- The CXX compiler identification is GNU 14.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- OpenROAD version: v2.0-17013-gf7f634f88
-- System name: Linux
-- Compiler: GNU 14.2.1
-- Build type: RELEASE
-- Install prefix: /usr/local
-- C++ Standard: 17
-- C++ Standard Required: ON
-- C++ Extensions: OFF
-- The C compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test C_COMPILER_SUPPORTS__-Wall
-- Performing Test C_COMPILER_SUPPORTS__-Wall - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wall
-- Performing Test CXX_COMPILER_SUPPORTS__-Wall - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-array-bounds
-- Performing Test C_COMPILER_SUPPORTS__-Wno-array-bounds - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-array-bounds
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-array-bounds - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-nonnull
-- Performing Test C_COMPILER_SUPPORTS__-Wno-nonnull - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-nonnull
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-nonnull - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-maybe-uninitialized
-- Performing Test C_COMPILER_SUPPORTS__-Wno-maybe-uninitialized - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-maybe-uninitialized
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-maybe-uninitialized - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-format-overflow
-- Performing Test C_COMPILER_SUPPORTS__-Wno-format-overflow - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-format-overflow
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-format-overflow - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-variable
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-variable - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-variable
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-variable - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-function
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-function - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-function
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-function - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-write-strings
-- Performing Test C_COMPILER_SUPPORTS__-Wno-write-strings - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-write-strings
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-write-strings - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-sign-compare
-- Performing Test C_COMPILER_SUPPORTS__-Wno-sign-compare - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-sign-compare
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-sign-compare - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-deprecated
-- Performing Test C_COMPILER_SUPPORTS__-Wno-deprecated - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-deprecated
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-deprecated - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-c++11-narrowing
-- Performing Test C_COMPILER_SUPPORTS__-Wno-c++11-narrowing - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-c++11-narrowing
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-c++11-narrowing - Failed
-- Performing Test C_COMPILER_SUPPORTS__-Wno-register
-- Performing Test C_COMPILER_SUPPORTS__-Wno-register - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-register
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-register - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-format
-- Performing Test C_COMPILER_SUPPORTS__-Wno-format - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-format
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-format - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-reserved-user-defined-literal
-- PerformingCMake Warning (dev) at src/CMakeLists.txt:175 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning at src/CMakeLists.txt:267 (message):
  spdlog: SPDLOG_FMT_EXTERNAL=ON


CMake Warning (dev) at src/dpl/CMakeLists.txt:41 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/cts/CMakeLists.txt:36 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/drt/CMakeLists.txt:49 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/utl/CMakeLists.txt:38 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

 Test C_COMPILER_SUPPORTS__-Wno-reserved-user-defined-literal - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-reserved-user-defined-literal
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-reserved-user-defined-literal - Failed
-- Performing Test C_COMPILER_SUPPORTS__-fpermissive
-- Performing Test C_COMPILER_SUPPORTS__-fpermissive - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-fpermissive
-- Performing Test CXX_COMPILER_SUPPORTS__-fpermissive - Success
-- Performing Test C_COMPILER_SUPPORTS__-x
-- Performing Test C_COMPILER_SUPPORTS__-x - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__-x
-- Performing Test CXX_COMPILER_SUPPORTS__-x - Failed
-- Performing Test C_COMPILER_SUPPORTS__c++
-- Performing Test C_COMPILER_SUPPORTS__c++ - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__c++
-- Performing Test CXX_COMPILER_SUPPORTS__c++ - Failed
-- Performing Test C_COMPILER_SUPPORTS__-std=c++17
-- Performing Test C_COMPILER_SUPPORTS__-std=c++17 - Failed
-- Performing Test CXX_COMPILER_SUPPORTS__-std=c++17
-- Performing Test CXX_COMPILER_SUPPORTS__-std=c++17 - Success
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-but-set-variable
-- Performing Test C_COMPILER_SUPPORTS__-Wno-unused-but-set-variable - Success
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-but-set-variable
-- Performing Test CXX_COMPILER_SUPPORTS__-Wno-unused-but-set-variable - Success
-- TCL library: /usr/lib/libtcl.so
-- TCL header: /usr/include/tcl.h
-- TCL readline library: /usr/lib/libtclreadline.so
-- TCL readline header: /usr/include
-- Found SWIG: /usr/bin/swig (found suitable version "4.2.1", minimum required is "4.0")
-- Using SWIG >= 4.2.1 -flatstaticmethod flag for python
-- Found Boost: /usr/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found version "1.86.0")
-- boost: 1.86.0
-- Found GTest: /usr/lib/cmake/GTest/GTestConfig.cmake (found version "1.15.2")
-- GTest: 1.15.2
-- Found Python3: /usr/include/python3.12 (found version "3.12.7") found components: Development Development.Module Development.Embed
-- Found ZLIB: /usr/lib/libz.so (found version "1.3.1")
-- Found Threads: TRUE
-- spdlog: 1.14.1
-- Found BISON: /usr/bin/bison (found version "3.8.2")
-- Found Doxygen: /usr/bin/doxygen (found version "1.12.0") found components: doxygen dot
-- STA version: 2.6.0
-- STA git sha: aa598a2f14c5c142e90391a69988523505e7db3d
-- System name: Linux
-- Compiler: GNU 14.2.1
-- Build type: RELEASE
-- Build CXX_FLAGS: -O3 -DNDEBUG
-- Install prefix: /usr/local
-- Found FLEX: /usr/bin/flex (found version "2.6.4")
-- TCL library: /usr/lib/libtcl.so
-- TCL header: /usr/include/tcl.h
-- TCL readline library: /usr/lib/libtclreadline.so
-- TCL readline header: /usr/include/tclreadline.h
-- CUDD library: /usr/local/lib/libcudd.a
-- CUDD header: /usr/local/include/cudd.h
-- SSTA: 0
-- Found SWIG: /usr/bin/swig (found suitable version "4.2.1", minimum required is "3.0")
-- STA executable: /home/ARCHIVE/OPENROAD/google-openroad-flow-scripts/tools/OpenROAD/src/sta/app/sta
-- Found re2: /opt/or-tools/lib/cmake/re2/re2Config.cmake (found version "11.0.0")
-- Found Clp: /opt/or-tools/lib/cmake/Clp/ClpConfig.cmake (found version "1.17.7")
-- Found Cbc: /opt/or-tools/lib/cmake/Cbc/CbcConfig.cmake (found version "2.10.7")
-- Found SCIP: /opt/or-tools/lib/cmake/scip/scip-config.cmake (found version "9.0.0")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Found OR-Tools: /opt/or-tools/lib/cmake/ortools (version: 9.11.4210)
-- TCL library: /usr/lib/libtcl.so
-- TCL header: /usr/include/tcl.h
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- GUI is enabled
-- Charts widget is not enabled
-- Found Boost: /usr/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found version "1.86.0") found components: serialization
-- Could NOT find VTune (missing: VTune_LIBRARIES VTune_INCLUDE_DIRS)
-- Found Boost: /usr/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found suitable version "1.86.0", minimum required is "1.78")
-- TCL library: /usr/lib/libtcCMake Warning (dev) at src/dst/CMakeLists.txt:42 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/dft/test/CMakeLists.txt:18 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

Number of processor cores: 16
l.so
-- TCL header: /usr/include/tcl.h
-- Found Boost: /usr/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found version "1.86.0") found components: serialization system thread
-- Found Boost: /usr/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found version "1.86.0")
-- Found Eigen3: /usr/share/eigen3/cmake/Eigen3Config.cmake (found version "3.4.0")
-- TCL readline enabled
-- Tcl Extended disabled
-- Python3 enabled
-- Configuring done (9.5s)
-- Generating done (0.8s)
-- Build files have been written to: /tmp/tmp.pvRoKTV8Mz
```

### To Reproduce

Default flow with ASAP7 Ibex used. Data races can be reproduced with:
* Run CTS for Ibex, just to prepare dependencies - `DESIGN_CONFIG=designs/asap7/ibex/config.mk make cts` (from `OpenROAD-flow-scripts/flow`)
* Rebuild OpenROAD with `TSAN` option - `./build_openroad.sh --local --openroad-args " -DTSAN=ON "` (from `OpenROAD-flow-scripts`)
* Run `5_1_grt` for Ibex - `DESIGN_CONFIG=designs/asap7/ibex/config.mk make do-5_1_grt` (from `OpenROAD-flow-scripts/flow`)

### Relevant log output

```shell
==================
WARNING: ThreadSanitizer: data race (pid=69109)
  Read of size 4 at 0x7ffd5673da24 by thread T16:
    #0 ant::AntennaChecker::checkAntennas(odb::dbNet*, int, bool) [clone ._omp_fn.0] /OpenROAD-flow-scripts/tools/OpenROAD/src/ant/src/AntennaChecker.cc:1117 (openroad+0x374885f)
    #1 <null> <null> (libgomp.so.1+0x1cd8d)

  Previous write of size 4 at 0x7ffd5673da24 by thread T23:
    #0 ant::AntennaChecker::checkAntennas(odb::dbNet*, int, bool) [clone ._omp_fn.0] /OpenROAD-flow-scripts/tools/OpenROAD/src/ant/src/AntennaChecker.cc:1117 (openroad+0x374886f)
    #1 <null> <null> (libgomp.so.1+0x1cd8d)
[...]
==================
[...]
==================
WARNING: ThreadSanitizer: data race (pid=69109)
  Write of size 8 at 0x7b1400be1230 by main thread:
    #0 operator delete(void*, unsigned long) ../../../../src/libsanitizer/tsan/tsan_new_delete.cpp:150 (libtsan.so.2+0x86c3d)
    #1 std::__new_allocator<std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> > >::deallocate(std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> >*, unsigned long) /usr/include/c++/12/bits/new_allocator.h:158 (openroad+0x3738724)
    #2 std::allocator_traits<std::allocator<std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> > > >::deallocate(std::allocator<std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> > >&, std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> >*, unsigned long) /usr/include/c++/12/bits/alloc_traits.h:496 (openroad+0x3738724)
    #3 std::_Rb_tree<odb::dbNet*, std::pair<odb::dbNet* const, ant::ViolationReport>, std::_Select1st<std::pair<odb::dbNet* const, ant::ViolationReport> >, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::_M_put_node(std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> >*) /usr/include/c++/12/bits/stl_tree.h:565 (openroad+0x3738724)
    #4 std::_Rb_tree<odb::dbNet*, std::pair<odb::dbNet* const, ant::ViolationReport>, std::_Select1st<std::pair<odb::dbNet* const, ant::ViolationReport> >, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::_M_drop_node(std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> >*) /usr/include/c++/12/bits/stl_tree.h:632 (openroad+0x3738724)
    [...]
    #37 tclAppInit /OpenROAD-flow-scripts/tools/OpenROAD/src/Main.cc:507 (openroad+0xd0ce43)
    #38 ord::tclInit(Tcl_Interp*) /OpenROAD-flow-scripts/tools/OpenROAD/src/Main.cc:550 (openroad+0xd0d9ae)
    #39 ord::tclAppInit(Tcl_Interp*) /OpenROAD-flow-scripts/tools/OpenROAD/src/Main.cc:545 (openroad+0xd0d9ae)
    #40 Tcl_MainEx <null> (libtcl8.6.so+0x108aa0)
    #41 __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58 (libc.so.6+0x27249)

  Previous read of size 8 at 0x7b1400be1230 by thread T15:
    #0 std::_Rb_tree<odb::dbNet*, std::pair<odb::dbNet* const, ant::ViolationReport>, std::_Select1st<std::pair<odb::dbNet* const, ant::ViolationReport> >, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::_S_left(std::_Rb_tree_node_base*) /usr/include/c++/12/bits/stl_tree.h:780 (openroad+0x374adfb)
    #1 std::_Rb_tree<odb::dbNet*, std::pair<odb::dbNet* const, ant::ViolationReport>, std::_Select1st<std::pair<odb::dbNet* const, ant::ViolationReport> >, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::_M_lower_bound(std::_Rb_tree_node<std::pair<odb::dbNet* const, ant::ViolationReport> >*, std::_Rb_tree_node_base*, odb::dbNet* const&) /usr/include/c++/12/bits/stl_tree.h:1952 (openroad+0x374adfb)
    #2 std::_Rb_tree<odb::dbNet*, std::pair<odb::dbNet* const, ant::ViolationReport>, std::_Select1st<std::pair<odb::dbNet* const, ant::ViolationReport> >, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::lower_bound(odb::dbNet* const&) /usr/include/c++/12/bits/stl_tree.h:1270 (openroad+0x374adfb)
    #3 std::map<odb::dbNet*, ant::ViolationReport, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::lower_bound(odb::dbNet* const&) /usr/include/c++/12/bits/stl_map.h:1307 (openroad+0x374adfb)
    #4 std::map<odb::dbNet*, ant::ViolationReport, std::less<odb::dbNet*>, std::allocator<std::pair<odb::dbNet* const, ant::ViolationReport> > >::at(odb::dbNet* const&) /usr/include/c++/12/bits/stl_map.h:549 (openroad+0x374adfb)
    #5 ant::AntennaChecker::checkGates(odb::dbNet*, bool, bool, std::basic_ofstream<char, std::char_traits<char> >&, odb::dbMTerm*, float, std::map<odb::dbITerm*, std::map<odb::dbTechLayer*, ant::NodeInfo, std::less<odb::dbTechLayer*>, std::allocator<std::pair<odb::dbTechLayer* const, ant::NodeInfo> > >, std::less<odb::dbITerm*>, std::allocator<std::pair<odb::dbITerm* const, std::map<odb::dbTechLayer*, ant::NodeInfo, std::less<odb::dbTechLayer*>, std::allocator<std::pair<odb::dbTechLayer* const, ant::NodeInfo> > > > > >&, std::vector<ant::Violation, std::allocator<ant::Violation> >&) /OpenROAD-flow-scripts/tools/OpenROAD/src/ant/src/AntennaChecker.cc:780 (openroad+0x3744c8f)
    #6 ant::AntennaChecker::checkNet(odb::dbNet*, bool, bool, std::basic_ofstream<char, std::char_traits<char> >&, odb::dbMTerm*, float, int&, int&, std::vector<ant::Violation, std::allocator<ant::Violation> >&) /OpenROAD-flow-scripts/tools/OpenROAD/src/ant/src/AntennaChecker.cc:1014 (openroad+0x3747f4c)
    #7 ant::AntennaChecker::checkAntennas(odb::dbNet*, int, bool) [clone ._omp_fn.0] /OpenROAD-flow-scripts/tools/OpenROAD/src/ant/src/AntennaChecker.cc:1122 (openroad+0x37487ab)
    #8 <null> <null> (libgomp.so.1+0x1cd8d)
[...]
```

### Screenshots

_No response_

### Additional Context

As ThreadSanitizer warnings can be quite long, logs are attached as file: [5_1_grt.log](https://github.com/user-attachments/files/17803129/ibex_tsan_5_1_grt.log)

### PR DESCRIPTION
**Pull request by luis201420**
This PR fixes data races found by ThreadSanitizer in ANT.  It is caused by the `net_to_report_` shared variable. 
The modifications made are:
- Adding mutex to protect access to the map.
- Removing unused parameters in functions.

Fixes #6186

### CODE REVIEW

**Comment by eder-matheus** [comment]
@luis201420 Please look at the failures in the CI. The regression tests are crashing.

File: src/ant/include/ant/AntennaChecker.hh [hunk 1]
```diff
@@ -243,6 +245,7 @@ class AntennaChecker
   std::string report_file_name_;
   std::vector<odb::dbNet*> nets_;
   std::map<odb::dbNet*, ViolationReport> net_to_report_;
+  std::mutex mapMutex;
```
Comment (eder-matheus) [inline src/ant/include/ant/AntennaChecker.hh:248]: Class members should follow the same name pattern: map_mutex_

**Review by eder-matheus — CHANGES_REQUESTED** [review]

**Review by eder-matheus — APPROVED** [review]

**Comment by luis201420** [comment]
The latest changes address and resolve all the issues identified by ThreadSanitizer.

File: src/ant/src/AntennaChecker.cc [hunk 2]
```diff
@@ -747,6 +754,7 @@ bool AntennaChecker::checkRatioViolations(odb::dbNet* db_net,
 
 void AntennaChecker::writeReport(std::ofstream& report_file, bool verbose)
 {
+  std::lock_guard<std::mutex> lock(map_mutex_);
```
Comment (eder-matheus) [inline src/ant/src/AntennaChecker.cc:757]: Why is this needed? writeReport doesn't run in multithread.

File: src/ant/src/AntennaChecker.cc [hunk 3]
```diff
@@ -756,6 +764,7 @@ void AntennaChecker::writeReport(std::ofstream& report_file, bool verbose)
 
 void AntennaChecker::printReport()
 {
+  std::lock_guard<std::mutex> lock(map_mutex_);
```
Comment (eder-matheus) [inline src/ant/src/AntennaChecker.cc:767]: Same question here. printReport is only called inside checkAntennas, out of a multithread region of the code.

**Review by eder-matheus — APPROVED** [review]

**Pull request merged**